solr/solrj/src/java/org/apache/solr/client/solrj/impl/HttpSolrClient.java

```diff
@@ -624,6 +641,12 @@ private HttpEntityEnclosingRequestBase fillContentStream(
       if (procCt != null) {
         String procMimeType = ContentType.parse(procCt).getMimeType().trim().toLowerCase(Locale.ROOT);
         if (!procMimeType.equals(mimeType)) {
+          if (isUnmatchedErrorCode(mimeType, httpStatus)) {
```

Comment:
I don't understand when this condition occurs. Can you add some comments explaining it?